### DISCUSSION TITLE
How do I fix this assertion failed amountweth != expectedweth

### DISCUSSION THREAD

**Discussion by 0x-codeX** [Cyfrin/foundry-full-course-cu#1627] [Q&A] [answered]
I know this has to do with the ERC20Mock and MockAggregator but I cannot seem to figure out how to get the priceFeed to provide the latest amount I assumed since I added the function in OracleLib it should work but I still get the same error message

Function
`function testGetTokenAmountFromUsd() public view {
        // If we want $100 of WETH @ $2000/WETH, that would be 0.05 WETH
        uint256 expectedWeth = 0.05 ether;
        uint256 amountWeth = dsce.getTokenAmountFromUsd(weth, 100 ether);
        assertEq(expectedWeth, amountWeth);
    }`

Error Message
` [29094] DSCEngineTest::testGetTokenAmountFromUsd()
    ├─ [18699] DSCEngine::getTokenAmountFromUsd(ERC20Mock: [0xBb2180ebd78ce97360503434eD37fcf4a1Df61c3], 100000000000000000000 [1e20]) [staticcall]
    │   ├─ [12663] OracleLib::staleCheckLatestRoundData() [delegatecall]
    │   │   ├─ [8993] MockV3Aggregator::latestRoundData() [staticcall]
    │   │   │   └─ ← 1, 700000000000 [7e11], 1, 1, 1
    │   │   └─ ← 1, 700000000000 [7e11], 1, 1, 1
    │   └─ ← 14285714285714285 [1.428e16]
    ├─ [0] VM::assertEq(50000000000000000 [5e16], 14285714285714285 [1.428e16]) [staticcall]
    │   └─ ← assertion failed: 50000000000000000 != 14285714285714285
    └─ ← assertion failed: 50000000000000000 != 14285714285714285

Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 6.65ms

Ran 1 test suite: 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/unit/DSCEngineTest.t.sol:DSCEngineTest
[FAIL. Reason: assertion failed: 50000000000000000 != 14285714285714285] testGetTokenAmountFromUsd() (gas: 29094)`

**Comment by EngrPips** [comment, marked as answer]
Hello @codexNature, The price that your `price feed` returned is `700000000000.` You claimed you had set the `initial price` of the `price feed` to `$2000/WETH.` Your `price feed` doesn't reflect that, so you check your `price feed` implementation and ensure nothing is implemented wrongly. Also, check your `helper config` to see if you didn't make any mistake when setting your price feed's `initial price` there. Here is a screenshot of me running the test and the output on my end.

<img width="1440" alt="Screenshot 2024-04-21 at 08 17 26" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/9d7883b7-54ad-44fa-891c-f1c361477f1d">

**Reply by 0x-codeX** [reply]
Thank you again @EngrPips I have been able to fix the issue, and in the process, I have a better understanding of how HelperConfigs works with DSCEngine and its connection with DSCEngineTest, it was a bulb lit up in my head. Thank you.

**Reply by EngrPips** [reply]
Awesome, I am thrilled you cracked it. Keep crushing it, friend.
